Source/ablastr/parallelization/MPIInitHelpers.cpp

```
@@ -75,6 +89,15 @@ namespace ablastr::parallelization
     {
 #ifdef AMREX_USE_MPI
         MPI_Finalize();
+
+        // INC0509050: Handle MPI finalization error on Tuolumne (LLNL)
+#ifdef AMREX_USE_HIP
+        hsa_status_t status = hsa_shut_down();
+        if (status != HSA_STATUS_SUCCESS) {
+            std::cerr << "hsa_shut_down failed with error code " << status << "! Aborting now.\n";
+            throw std::runtime_error("hsa_shut_down failed after MPI_Finalize completed.");
```

Do we want to throw a runtime error or handle the exception differently?